File: src/main/java/org/shark/example/service/emqx/SendMessageService.java
package org.shark.example.service.emqx;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.shark.example.service.base.pojo.ResponseDto;
import org.shark.example.service.mqtt.MqttUseCase;
import org.springframework.stereotype.Service;

@Slf4j
@RequiredArgsConstructor
@Service
public class SendMessageService {

    private final MqttUseCase mqttUseCase;

    public ResponseDto<Void> start(String message) {
        mqttUseCase.pushMessage(message);
        return ResponseDto.<Void>builder()
                .status(true)
                .build();
    }
}

File: src/main/kotlin/org/shark/example/service/emqx/SendMessageService.kt
package org.shark.example.service.emqx

import org.shark.example.service.base.pojo.ResponseDto
import org.shark.example.service.mqtt.MqttUseCase
import org.springframework.stereotype.Service
import org.slf4j.LoggerFactory

@Service
class SendMessageService(
    private val mqttUseCase: MqttUseCase
) {
    private val log = LoggerFactory.getLogger(SendMessageService::class.java)

    fun start(message: String): ResponseDto<Void> {
        mqttUseCase.pushMessage(message)
        return ResponseDto(
            status = true
        )
    }
}
